### ISSUE TITLE
MockFS File.Copy throws if a file handle with ReadWrite share is open 

### ISSUE THREAD

**Issue by AnakinRaW** [Testably/Testably.Abstractions#549]
The following (simplified) code fails with the test FS when it actually works on the OS's FS. 

```cs
_fileSystem.Initialize() .WithFile("test/1.txt").Which(f => f.HasStringContent("1"))
var fsStream = _fileSystem.FileStream.New("test/1.txt", FileMode.Open, FileAccess.Read, FileShare.Read);
_fileSystem.File.Copy("test/1.txt", "other/1.txt", true);
fs.Dispose();
```
Also fails if share is ReadWrite.

If my brief debugging is correct this is caused by the Mock copy operation does not allow any sharing.

**Comment by vbreuss** [comment]
@AnakinRaW 
Thanks for reporting the issue. I could reproduce it and am working on a fix in #550. 

Please note, that the behaviour seems to differ between Windows and Linux/MacOS...

**Comment by AnakinRaW** [comment]
Even the issue is closed, i found this: https://github.com/dotnet/runtime/issues/52700
might be interesting...

**Comment by vbreuss** [comment]
> Even the issue is closed, i found this: [dotnet/runtime#52700](https://github.com/dotnet/runtime/issues/52700) might be interesting...

@AnakinRaW:
After reading the dotnet-issue, I also fixed the behavior for `File.Move` and `File.Replace` in #555.
